### ISSUE TITLE
Error enabling Flyspell mode

### ISSUE THREAD

**Issue by QikeLi** [jkitchin/scimax#32]
I got an error message anytime I try to turn on flyspell mode after I run (org-agenda-to-appt):

    Error enabling Flyspell mode:(ispell-phaf: No matching entry for en_US in ‘ispell-hunspell-dict-paths-alist’.)

However, no error will occur if I turn on flyspell mode first and then run (org-agenda-to-appt). Any suggestions how to solve this issue?

Thank you very much!

**Comment by jkitchin** [comment]
Are you using windows?

**Comment by QikeLi** [comment]
I use  Mac OSX EL Capitan Version 10.11.6

**Comment by jkitchin** [comment]
hm. 
My ispell-hunspell-dict-paths-alist is set to:

(("american" "/Users/jkitchin/Library/Spelling/en_US.aff") ("british"
 "/Users/jkitchin/Library/Spelling/en_GB.aff") ("english"
 "/Users/jkitchin/Library/Spelling/en_US.aff") ("en_ZA"
 "/Users/jkitchin/Library/Spelling/en_ZA.aff") ("en_US"
 "/Users/jkitchin/Library/Spelling/en_US.aff") ("en_GB"
 "/Users/jkitchin/Library/Spelling/en_GB.aff") ("en_CA"
 "/Users/jkitchin/Library/Spelling/en_CA.aff") ("en_AU"
 "/Users/jkitchin/Library/Spelling/en_AU.aff") ("default"
 "/Users/jkitchin/Library/Spelling/default.aff"))

Maybe you can find where en_US.aff is on your computer and than 
(add-to-list 'ispell-hunspell-dict-paths-alist '("en_US"  "your-path-to/en_US.aff")

**Comment by prtkm** [comment]
I remember had to download the dictionary from a libreoffice server and put it in a Spelling folder on my Mac

**Comment by QikeLi** [comment]
Thank you guys for the responses! I did download the dictionary and put it in ~/Library/Spelling.

Maybe I should restate the behavior:
If I manually enable flyspell-mode right after I open emacs, flyspell works fine and ispell-hunspell-dict-paths-alist contains the path "~/Library/Spelling". But If I open emacs and then run (org-agenda-to-appt) or open a .org file, I would see the error message:

        Error enabling Flyspell mode:(ispell-phaf: No matching entry for en_US in ‘ispell-hunspell-dict-paths-alist’.)

An update of the behavior, it seems opening a .org file will also trigger the  error message not just the command (org-agenda-to-appt).

**Comment by jkitchin** [comment]
Are you using scimax? If you are, then flyspell-mode should be getting turned on in an org-mode hook.  This is setup when flyspell-correct-ivy is loaded.

**Comment by QikeLi** [comment]
Yes, I am using scimax.

I solved the problem by adding the following line in my scimax/user/user.el file so that flyspell-mode is forced to be turned on before I open any .org files.

  	 (flyspell-mode)

Apparently, it is an ad hoc solution. I wonder if the problem was caused by the fact that I put some my personal configurations in the scimax/packages.el file. Is this bad practice?  Thank you!

**Comment by dph3dg** [comment]
I'm having similar problems. I can set the correct dictionary in my user/preload.el but init.el then resets it to the default dictionary. What is the best way to counteract this protocol, in customised variables?

**Comment by Cumol** [comment]
Copying this from the other issue I posted:

I am trying to set up hunspell with the en_US dictionary.

every time I open a project with projectile I get the following message in my mini-buffer
(ispell-phaf: No mathcing entry for en_US in 'ispell-hunspell-dict-paths-alist".)

I installed hunspell via brew with:

brew install hunspell

and when I check hunspell -D I get the following:
AVAILABLE DICTIONARIES (path is not mandatory for -d option): ./en_US Library/Spelling/en_US Can't open affix or dictionary files for dictionary named "default".

Now I thought of adding the link to the .aff file as suggested by Alex Gromnitsky like this:

(setq ispell-hunspell-dict-paths-alist '(("en_US" "~/Library/Spelling/en_US.aff")

without success.

Now I put (flyspell-mode) into the user.init file and get rid of the original error message, however, I now get the following:

`Error enabling Flyspell mode:(ispell-phaf: No matching entry for en_US in ‘ispell-hunspell-dict-paths-alist’.)`

Which is the same as what @QikeLi gets.

**Comment by jkitchin** [comment]
It sounds like hunspell is not installed completely. When I run this:
$ hunspell -D

I get:

SEARCH PATH:
.::/usr/share/hunspell:/usr/share/myspell:/usr/share/myspell/dicts:/Library/Spelling:/Users/jkitchin/.openoffice.org/3/user/wordbook::/Users/jkitchin/.openoffice.org2/user/wordbook::/Users/jkitchin/.openoffice.org2.0/user/wordbook::/Users/jkitchin/Library/Spelling:/opt/openoffice.org/basis3.0/share/dict/ooo:/usr/lib/openoffice.org/basis3.0/share/dict/ooo:/opt/openoffice.org2.4/share/dict/ooo:/usr/lib/openoffice.org2.4/share/dict/ooo:/opt/openoffice.org2.3/share/dict/ooo:/usr/lib/openoffice.org2.3/share/dict/ooo:/opt/openoffice.org2.2/share/dict/ooo:/usr/lib/openoffice.org2.2/share/dict/ooo:/opt/openoffice.org2.1/share/dict/ooo:/usr/lib/openoffice.org2.1/share/dict/ooo:/opt/openoffice.org2.0/share/dict/ooo:/usr/lib/openoffice.org2.0/share/dict/ooo
AVAILABLE DICTIONARIES (path is not mandatory for -d option):
/Users/jkitchin/Library/Spelling/default
/Users/jkitchin/Library/Spelling/en_AU
/Users/jkitchin/Library/Spelling/en_CA
/Users/jkitchin/Library/Spelling/en_GB
/Users/jkitchin/Library/Spelling/en_US
/Users/jkitchin/Library/Spelling/en_ZA
/Users/jkitchin/Library/Spelling/hyph_en_GB
/Users/jkitchin/Library/Spelling/hyph_en_US
LOADED DICTIONARY:
/Users/jkitchin/Library/Spelling/en_US.aff
/Users/jkitchin/Library/Spelling/en_US.dic

Are you using the emacs inside of scimax? Or an emacs of your own?

If you are using your own, you need to make sure these files:
https://github.com/jkitchin/emacs-win/tree/master/share/hunspell
are installed somewhere.

Then, I do something like this in scimax:
```
(if (file-directory-p (expand-file-name "emacs-win" scimax-dir))
      (progn
	;; spell-checking on windows
	(setq ispell-program-name
	      (expand-file-name
	       "emacs-win/bin/hunspell"
	       scimax-dir))

	(setq ispell-dictionary "english")

	(setq ispell-local-dictionary-alist
	      `(("english"
		 "[[:alpha:]]"
		 "[^[:alpha:]]"
		 "[']"
		 t
		 ("-d" "en_US" "-p" ,(expand-file-name
				      "emacs-win/share/hunspell/en_US"
				      scimax-dir))
		 nil
		 utf-8))))
    (setq ispell-program-name (executable-find "hunspell")
	  ispell-dictionary "en_US"
	  flyspell-correct-interface 'flyspell-correct-ivy))
```

I only do that for the scimax installed by the scimax windows install script, which adds the emacs-win sub-repository to scimax that contains those files, so If you are not using that emacs, and are on windows, you might need something like that.

**Comment by Cumol** [comment]
I finally made it work. Emacs used to underline every second word as a spelling mistake but lack to show any correction suggestions. Now it works as it should.

I had to force uninstall all versions of aspell I had installed with Homebrew:
`brew uninstall --force aspell`

Then reinstalled aspell with a english dictionary:
brew install aspell --with-lang-en

And then added this in my user.el file:

```
 (setq ispell-program-name "aspell")
 ;; Please note ispell-extra-args contains ACTUAL parameters passed to aspell
 (setq ispell-extra-args '("--sug-mode=ultra" "--lang=en_US")
```

I don't know why this is working now though, maybe you have an idea @jkitchin ...
